### ISSUE TITLE
Is there any way to decrypt the replay binary file?

### ISSUE THREAD

**Issue by kwuenkrab** [crazydoomy/MD-Replay-Editor#2]
Thanks so much for your work. It works properly in the game.
I open the .replay file with notepad++. It seems that some words make sense when converted to ASCII. While most of them is still unfathomable. I'm wondering if there's a way to decrypt all of them? Then I don't have to open the game to load the replay.

**Comment by kwuenkrab** [comment]
![samples_core_comparison](https://user-images.githubusercontent.com/97434237/174486324-65c4af8a-b5e6-4a24-86af-d80ed2cefb9c.png)
I found the codes starting with "gqFixQ"/ "677146697851" is a sequence of HEX numbers. Then I searched some repeated numbers like 75/75/67 and highlight them with the searching tool in Vim. I hope this will be helpful.

**Comment by crazydoomy** [comment]
I have tried decrypting the replay file itself, there is a sequence of base64 but I haven't been able to figure out anything from it.

**Comment by kwuenkrab** [comment]
![samples_comparison](https://user-images.githubusercontent.com/97434237/174512139-909f469e-600e-4240-8f5d-b9675c7d63a9.png)
I compare 2 samples and found there're some common subsequences between them.

**Comment by crazydoomy** [comment]
the start of them are similar, some number of bytes in there is the length of the file, and there is common stuff like player names, deck, and avatars.

**Comment by kwuenkrab** [comment]
Right. I've found that the opponent's deck info is also included. I'm currently looking into an numeron's extra deck and there're some consecutive numbers, which matches the pattern of "Numeron Gates". But the ID is not the same as the ID from this sheet: https://docs.google.com/spreadsheets/d/1IXpwCaabi47Ly8dAf4aJtFdYwi29yGkCJ4bvMhnsvc8

**Comment by kwuenkrab** [comment]
Actually, the ID is matched. There is no offset.
For example, in the codes"ExtraC40000449FCD3C1403001415030014160300141703007113CD3841CD37D97700149F55000376005FA4", where "3C1403 00141503 00141603 00141703" is the IDs and numbers of the four "Numeron gates". In the sheet, the ID of the No.1 Numeron Gate is 15380/0x3c14.
But the following 001415 001416 001417 is a bit confusing. Why is it represented as 0014 instead of just 0015?

In addition, the 007113 indicates the ID of CNo1 numeron gate, whose ID in the sheet is 15379/0x3c13. But the amount is only one, hence it's different from the previous ones. CD3841CD is the ID of 14401/"Infinitrack Fortress Megaclops" and 37D977 is the ID of 14297/双穹の騎士アストラム. I've checked the extra deck in the game and it's completely matched.
The rest bits' function is unknown. Maybe it indicates the rest of the extra deck is empty.
Moreover, there is a "SIDE" in the following ASCII codes. I think it means that Konami will add a Bo3 game in the future.

**Comment by pixeltris** [comment]
> I have tried decrypting the replay file itself, there is a sequence of base64 but I haven't been able to figure out anything from it.

It's `base64(messagepack(Dictionary<string, object>))`

Where the dictionary is:

```
Dictionary<string, object>
{
 ["b"] = zlib(byte[]/replayData)
 ["f"] = { int/p1DuelFinishType, int/p2DuelFinishType }
}
```

https://github.com/pixeltris/YgoMaster/blob/2e9e9cf4ff63877c2230cd2368dafd93c172c9a8/YgoMasterClient/DuelStarter.cs#L275-L295

The duel records are 8 bytes each. Just above that highlighted code shows the insertion of surrendering (which was required for me to save solo duels as the client skips the normal surrender sequence).

**Comment by crazydoomy** [comment]
thank you for this
